docs/overview.md

```diff
@@ -1,54 +1,120 @@
+implement its service.
+
+A **Game Room Unity (GRU)** is where the game server logic will run and clients
+will connect to execute their matches. It's the most atomic entity in a Maestro
+architecture.
+
+One could define a GRU as a Kubernetes service (type nodePort) associated with a
+single pod. This restriction is made because in AWS we cannot load balance UDP.
+We're using containerized applications and Kubernetes in order to simplify the
+management and scaling of the game rooms.
+
+A **scheduler** is a specification for a set of GRUs. It defines the image,
+variables, computing and networks resources, permissions and other
+configurations for these GRUs. One can define them as a Kubernetes deployment
+strongly attached to a namespace. This restriction is made because Maestro does
+not allow more than one schedulers to be defined in the same namespace.
 
 ## Architecture
 
-Maestro is a game room scheduler that is composed by a controller, a watcher, a worker, an API and a CLI. In the future we may have an UI for displaying metrics such as:
+Here, we provide an overview of the service architecture. Maestro requires that
+these entities, except Postgres and Redis, run in the same Kubernetes cluster.
+
+The diagram below depicts the integration between services and we provide a
+description of each component responsibility.
 
-- % of rooms usage
-- rate of room occupation increase/decrease
-- rooms cpu and memory usage
-- etc.
+![diagram](./architecture.jpg "Maestro Architecture")
 
-### maestro-controller
+### API
 
-The controller is responsible for managing the Game Room Unities (GRUs). It creates and gracefully terminates GRUs according to auto scaling policies defined by the user. It makes use of the Kubernetes cluster's API endpoints in order to have updated information about the GRUs managed by Maestro. It is also responsible for persisting relevant information in the database and managing rooms statuses.
+The API is the connection of Maestro to the external world and with the GRUs
+themselves. It is responsible for:
 
-### maestro-watcher
+- Receiving and processing client operations requests over schedulers;
+- Listening to [GRU status](#room-protocol), through healthchecks;
+- Storing scheduler configurations in the database. They will be consulted by
+  the other entities, like the `watcher`;
+- Managing the pool of GRUs, keeping track of each host IP and port.
 
-The watcher ensures that at any time the Game Room Unities (GRUs) state is as expected. If the scaling policies say that one should have 10 GRUs of a given type, the watcher will ask the controller to create or terminate GRUs as needed. The desired state is kept in a database that is consulted by the watcher (via controller) each time it runs. It has a lock so Maestro can be scaled horizontally. Each scheduler (i.e. maestro scalable entity) has its own watcher.
+Then, we have **maestro-worker**. They are instanced on demand, 
```

Comment:
Right now we don't have any concept of creating workers on demand. We already had some discussions about using a different model but we already have to have at least one worker running and recently we are incentivising only having one instance of the worker.

Reply:
Changed it. PTAL?